src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/GeoServerRESTImporter.java

```diff
@@ -0,0 +1,701 @@
+package de.terrestris.shogun2.importer;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpException;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.entity.ContentType;
+import org.apache.log4j.Logger;
+import org.geotools.referencing.CRS;
+import org.geotools.referencing.wkt.Formattable;
+import org.opengis.referencing.FactoryException;
+import org.opengis.referencing.NoSuchAuthorityCodeException;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import de.terrestris.shogun2.importer.communication.AbstractRESTEntity;
+import de.terrestris.shogun2.importer.communication.RESTData;
+import de.terrestris.shogun2.importer.communication.RESTImport;
+import de.terrestris.shogun2.importer.communication.RESTImportTask;
+import de.terrestris.shogun2.importer.communication.RESTImportTaskList;
+import de.terrestris.shogun2.importer.communication.RESTLayer;
+import de.terrestris.shogun2.importer.communication.RESTTargetDataStore;
+import de.terrestris.shogun2.importer.communication.RESTTargetWorkspace;
+import de.terrestris.shogun2.importer.transform.RESTGdalAddoTransform;
+import de.terrestris.shogun2.importer.transform.RESTGdalTranslateTransform;
+import de.terrestris.shogun2.importer.transform.RESTGdalWarpTransform;
+import de.terrestris.shogun2.importer.transform.RESTReprojectTransform;
+import de.terrestris.shogun2.importer.transform.RESTTransform;
+import de.terrestris.shogun2.util.http.HttpUtil;
+import de.terrestris.shogun2.util.model.Response;
+import net.lingala.zip4j.core.ZipFile;
+import net.lingala.zip4j.exception.ZipException;
+import net.lingala.zip4j.model.FileHeader;
+import net.lingala.zip4j.model.ZipParameters;
+import net.lingala.zip4j.util.Zip4jConstants;
+
+/**
+ *
+ * @author Daniel Koch
+ * @author terrestris GmbH & Co. KG
+ *
+ */
+@Component
+public class GeoServerRESTImporter {
+
+	/**
+	 * The Logger.
+	 */
+	private final static Logger LOG = Logger.getLogger(GeoServerRESTImporter.class);
+
+	/**
+	 *
+	 */
+	private String username;
+
+	/**
+	 *
+	 */
+	private String password;
+
+	/**
+	 *
+	 */
+	private ObjectMapper mapper = new ObjectMapper();
+
+	/**
+	 *
+	 */
+	private URI baseUri;
+
+	/**
+	 *
+	 */
+	public GeoServerRESTImporter() {
+
+	}
+
+	/***
+	 *
+	 * @param importerBaseURL
+	 * @param defaultSRS
```

Comment:
not needed

Reply:
Fixed in b119a96e9828b8def01eddd0fd0dd3ede1fe207a.